docs/networking/simpleimpl.md

```diff
@@ -34,15 +33,33 @@ Returning `false` for both means that this channel must be present on the other
 Registering Packets
 -------------------
 
-Next, we must declare the types of messages that we would like to send and receive. This is done using `INSTANCE#registerMessage`, which takes 5 parameters:
+Next, we must declare the types of messages that we would like to send and receive. This is done using the following steps:
 
-- The first parameter is the discriminator for the packet. This is a per-channel unique ID for the packet. We recommend you use a local variable to hold the ID, and then call registerMessage using `id++`. This will guarantee 100% unique IDs.
-- The second parameter is the actual packet class `MSG`.
-- The third parameter is a `BiConsumer<MSG, FriendlyByteBuf>` responsible for encoding the message into the provided `FriendlyByteBuf`.
-- The fourth parameter is a `Function<FriendlyByteBuf, MSG>` responsible for decoding the message from the provided `FriendlyByteBuf`.
-- The final parameter is a `BiConsumer<MSG, Supplier<NetworkEvent.Context>>` responsible for handling the message itself.
+- First: Invoke the SimpleChannel#messageBuilder to starting a building of a message.
+- Then: Call `decoder`, `encoder` and `consumerMainThread` in any order to build your message.
+    The parameter of `decoder` is a `Function<FriendlyByteBuf, MSG>` responsible for decoding the message from the provided `FriendlyByteBuf`.
+    The parameter of `encoder` is a  `BiConsumer<MSG, FriendlyByteBuf>` responsible for encoding the message into the provided `FriendlyByteBuf`.
+    The parameter of `consumerMainThread` is a `BiConsumer<MSG, Supplier<NetworkEvent.Context>>` responsible for handling the message itself on the correct thread designated by `messageBuilder`.
+- Finally: call `add` to register the message to SimpleChannel.
+A confortable format: 
+```java
+private static final String PROTOCOL_VERSION = "1";
+public static final SimpleChannel INSTANCE = ChannelBuilder.named("mymodid","main"))
+            .clientAcceptedVersions(PROTOCOL_VERSION::equals)
+            .serverAcceptedVersions(PROTOCOL_VERSION::equals)
+            .acceptedVersions(PROTOCOL_VERSION::equals)
+            .optional().simpleChannel()
+              //register message
+            .messageBuilder(MyMessage.class, NetworkDirection.PLAY_TO_SERVER)
+            .decoder(buf ->{new MyMessage())
+            .encoder((message, buf) -> {})
```

Comment:
Should have `// ...` to show something is going on. No one should send an empty message.